src/hotspot/share/c1/c1_Runtime1.cpp

```diff
@@ -280,6 +285,7 @@ void Runtime1::initialize(BufferBlob* blob) {
 #endif
   BarrierSetC1* bs = BarrierSet::barrier_set()->barrier_set_c1();
   bs->generate_c1_runtime_stubs(blob);
```

Comment:
Don't we need to handle failures in generate_c1_runtime_stubs?  With the assert removed, I think we'll get a nullptr crash.

Reply:
Yep, you're right. Actually that call could potentially fail too.
I've added code to handle that case as well.
Thanks @dean-long!